sdk/src/metrics/meter.cc

```diff
@@ -52,6 +61,14 @@ nostd::unique_ptr<metrics::Counter<double>> Meter::CreateDoubleCounter(
     nostd::string_view description,
     nostd::string_view unit) noexcept
 {
+  if (!ValidateInstrument(name, description, unit))
+  {
+    OTEL_INTERNAL_LOG_ERROR("Meter::CreateDoubleCounter - failed. Invalid parameters."
```

Comment:
nit: probably use macro `__func__` to avoid hard code function name in string liternal.

Reply:
yes, probably we can modify `OTEL_INTERNAL_*` macros to automatically add class and method name to the logs.